Use HTTPS for links

Marcono1234 opened the pull request:
Currently there are multiple links in this project which use HTTP as protocol instead of HTTPS despite the target site supporting it. Even though the browser then upgrades the connection to HTTPS, it first sends an unencrypted request which could be read and whose response could be manipulated by an adversary in the same network.

Note that the creation of these changes was mostly automated, but only for links where a HTTPS connection did not appear to cause any issues.
Dead links (there are a few) and pages which returned a non-200 or 300 HTTP status code have not been changed; however I did not verify that every link still points to the correct page (in case a 200 status code was returned).

I have created a separate commit (e9b4c76) for the `License.rtf` file because I am not familiar with the RTF format and not sure whether simply replacing the link might cause issues (trying to edit the file in LibreOffice would have caused major changes). If these changes are fine I would squash the commit.

Please let me know if you want any changes or if you do not think that these changes are worth it, especially if the license header should not be changed (commit 595f5c2 of this pull request). Any feedback is appreciated!

Marcono1234 committed:
Use HTTPS for links

Marcono1234 commented on src/interactive/scala/tools/nsc/interactive/Pickler.scala:
```diff
@@ -33,8 +33,8 @@ import java.io.Writer
  *
- *  <a href="http://code.google.com/p/gdata-scala-client/wiki/DevelopersGuide">
- *  http://code.google.com/p/gdata-scala-client/wiki/DevelopersGuide
+ *  <a href="https://code.google.com/p/gdata-scala-client/wiki/DevelopersGuide">
+ *  https://code.google.com/p/gdata-scala-client/wiki/DevelopersGuide
  *  </a>
```
I am not familiar with Scala, but could this link be changed like this?
```suggestion
 * [[https://code.google.com/p/gdata-scala-client/wiki/DevelopersGuide]]
```

SethTisue commented:
thank you for this. can you sign the CLA, please?

Marcono1234 commented:
I have signed the CLA now. Should I squash commit e9b4c76 with 1446879 then? That commit was only temporary and the message "RISKY Use HTTPS for links" is not very useful.

Also could you please improve your CLA signing page? It appears (unless I overlooked it) that the page does not describe at all:
- Why full name and e-mail address are needed
- How full name and e-mail address will be stored / to whom they will be visible

NthPortal commented:
> Should I squash commit e9b4c76 with 1446879

yes please

> Can you alternatively scan and sign the CLA, or will I additionally receive a CLA form now which I have to sign?

it's an alternative if you don't want to use your github account

Marcono1234 commented:
Squashed them and rebased onto `scala:2.13.x`. Sorry for all the trouble this pull request is causing, maybe it would be best to omit commit 9d76aad? Because the value this adds very likely does not justify the amount of changes.

SethTisue commented:
> maybe it would be best to omit commit 9d76aad?

yes please — I agree it isn't worth the clutter in the version control history of so many files

SethTisue commented:
also I think it's better, on the balance, to leave `LICENSE` and `LICENSE.md` and `LICENSE.rtf` and `doc/licenses/apache_jna.txt` and `spec/public/octicons/LICENSE.txt` and `toc.js` alone, as those files are taken verbatim from upstream sources

SethTisue commented:
> Also could you please improve your CLA signing page? It appears (unless I overlooked it) that the page does not describe at all:
> Why full name and e-mail address are needed
> How full name and e-mail address will be stored / to whom they will be visible

The only texts I'm aware of that address this area of questioning are:

* https://www.lightbend.com/contribute/cla/scala
* https://contributors.scala-lang.org/t/moving-to-the-apache-v2-0-license/1859

My informal answers to your questions, as engineer on the project and not as management:

* I suppose it's so we can get in touch with you if there's some legal issue with your contribution? For example, if we were to discover that code you submitted had been taken from somewhere without proper attribution/licensing. Is this not common on other OSS projects?
* They're stored on a server and are not visible to anyone besides the copyright holders, namely EPFL and Lightbend.

cc @adriaanm

Marcono1234 commented:
Thanks; I have now omitted the commit changing the license headers and other license files.
